### PR TITLE
feat: Added source bundles to more symbol server sources

### PR DESCRIPTION
**Pull request by mitsuhiko**
This adds support for looking up source bundles from more sources.

### CODE REVIEW

File: docs/advanced/symbol-server-compatibility.md [hunk 1]
```diff
@@ -139,7 +173,17 @@ Casing rules for Symbol Server are mixed:
 The presence of a `index2.txt` in the root indicates two tier structure where
 the first two characters are prepended to the path as an additional folder. So
 `foo.exe/542D5742000f2000/foo.exe` is stored as
-`fo/foo.exe/542D5742000f2000/foo.exe`.
+`fo/foo.exe/542D5742000f2000/foo.exe`. Note that symbolicator does not probe
+for the `index2.txt` file. You need to be explicit in configuring it.
+
+Symbol bundles are only supported for PE/PDB files with the following format:
```
Comment (jan-auer) [inline docs/advanced/symbol-server-compatibility.md]:
```suggestion
Source bundles are only supported for PE/PDB files with the following format:
```

Might wanna clarify here that the source bundle is always keyed after the PDB and never the EXE.

File: docs/api/index.md [hunk 2]
```diff
@@ -84,26 +84,28 @@ These are common parameters that work on most symbol sources (except `sentry`):
 
 - `filters`: a set of filters to reduce the number of unnecessary hits on a
   symbol server. This configuration key is an object with two keys:
-    - `filetypes`: a list of file types to restrict the server to. Possible
-      values: `pe`, `pdb`, `mach_debug`, `mach_code`, `elf_debug`, `elf_code`,
-      `breakpad`)
-    - `path_patterns`: a list of glob matches that need to be matched on the image
-      name. If the debug image has no name it will never match here.
+  - `filetypes`: a list of file types to restrict the server to. Possible
```
Comment (jan-auer) [inline docs/api/index.md]: I think this screwed up indentation in the generated documentation if only two spaces are used.

File: src/utils/paths.rs [hunk 3]
```diff
@@ -154,7 +154,9 @@ fn get_native_path(filetype: FileType, identifier: &ObjectId) -> Option<String>
                 ObjectType::Elf => get_gdb_path(identifier)?,
                 ObjectType::Unknown => return None,
             };
-            base_path.truncate(base_path.len() - base_path.rsplitn(2, '.').next()?.len() - 1);
+            base_path.truncate(
+                (base_path.len() - base_path.rsplitn(2, '.').next()?.len()).checked_sub(1)?,
```
Comment (jan-auer) [inline src/utils/paths.rs]: pretty sure what you want here is:

```rust
if let Some(cutoff) = base_path.rfind('.') {
    base_path.truncate(cutoff);
}
```

Comment (jan-auer) [reply src/utils/paths.rs]: Also, probably this should protect against dots in the path, since file names may contain dots. Paths are hard :D

Comment (jan-auer) [reply src/utils/paths.rs]: For copy/paste convenience:

```rust
            if let Some(cutoff) = base_path.rfind('.') {
                if cutoff > base_path.rfind('/').unwrap_or(0) {
                    base_path.truncate(cutoff);
                }
            }
```

(note that this changes test assertions, rightfully so.

**Comment by mitsuhiko** [comment]
This was only supposed to remove the extensions from pdb files. Changed the code.

**Pull request merged**